CMakeLists.txt

```diff
@@ -1217,6 +1217,34 @@ else ()
   set(HAVE_HALF_FLOAT OFF)
 endif ()
 
+# Define an macro to check if we have include and initialized libipt.
+set (HAVE_LIBIPT OFF)
+# Initialize libipt and active the macro HAVE_LIBIPT
+if (LINUX AND X64)
+  find_package(Git QUIET)
+  if (GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+  # Update submodules as needed
+    option(GIT_SUBMODULE "Check submodules during build" ON)
+    if (GIT_SUBMODULE)
+      message(STATUS "Submodule update")
+      execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
+                        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+                        RESULT_VARIABLE GIT_SUBMOD_RESULT)
+      if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+          message(STATUS
+          "git submodule update --init --recursive failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+      else()
+        # Add libipt to build system. For kernel traces decode, we need to active SIDEBAND, PEVENT and FEATURE_ELF
+        set(SIDEBAND ON CACHE BOOL "Enable libipt-sb, a sideband correlation library" FORCE)
```

Comment:
So libipt does *not* use the CMake export mechanism for modular separation and we have to smoosh it into ours?  I see from these option names that it does not have them all prefixed for isolation: so we are at risk of CMake name collisions and subtle problems.  Worth a comment that this is not the ideal.